### ISSUE TITLE
False negative: expected undefined-variable when deleted variable used

### ISSUE THREAD

**Issue by jaraco** [PyCQA/pylint#7460]
### Bug description

While working on a piece of code, `pylint` warned about an unused variable from a function parameter. Especially because the name/position of the function argument is important to the interface, the recommended "fix" for the warning was to delete the unused variable. Later changes to that function indicated the use of that variable, so it became used again. However, at this point, the linter was happy to allow this code pass until runtime.

```
 draft $ cat a.py
# pylint: disable=missing-module-docstring, disable=disallowed-name, disable=missing-function-docstring
def simplefunc(abc):
    del abc  # unused
    print("hello world", abc)
 draft $ pip-run -q pylint -- -m pylint a.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

 draft $ py -c "import a; a.simplefunc(None)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/jaraco/draft/a.py", line 4, in simplefunc
    print("hello world", abc)
                         ^^^
UnboundLocalError: cannot access local variable 'abc' where it is not associated with a value
```

### Command used

```shell
python -m pylint a.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

Expected `undefined-variable` or `possibly-undefined-variable` to trigger.

### Pylint version

```shell
pylint 2.15.2
astroid 2.12.9
Python 3.10.6 (main, Aug 30 2022, 04:58:14) [Clang 13.1.6 (clang-1316.0.21.2.5)]
```

**Comment by jacobtylerwalls** [comment]
Same root cause as #841, `pylint` lacks control-flow awareness of `del`.
